diagnostic.md

```diff
@@ -5,74 +5,90 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```md
-// your response here
+A backend's purpose is to handle and respond to requests from the client, or front end.  It handles things like business logic and database interactions.  The back end essentially enables the front end experience.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```md
-// your response here
+Model
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```md
-// your response here
+Controller
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```md
-// your response here
+Views are the presentation layer and are essentially visual representations of models.  They render the models into one or more formats, such as XML.
 ```
 
 What does C.R.U.D stand for?
 
 ```md
-// your response here
+Create
+Read
+Update
+Destroy
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```md
-// your response here
+They can be found within the controller.
 ```
 
 List at least 5 standard rails actions that C.R.U.D requests correspond to?
 
 ```md
-// your response here
+Index
+Show
+Create
+Update
+Destroy
 ```
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```md
-// your response here
+-Request is sent to the web server
+-The router directs the request to the people controller
+-The controller directs the request to the appropriate model
+-The model returns data to the controller
+-The controller returns data to the web server
+-The web server displays a response for the user
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+rails new
 ```
 
 What is the command to start an instance of a rails server?
 
 ```bash
-// your response here
+bundle exec rails server
 ```
 
 What are the commands to drop, create, migrate and seed a database from the command
 line? (5 bullet points)
 
 ```bash
-// your response here
+db:drop
+db:create
+db:seed
+db:setup
+db:reset
 ```
 
 What is the command to scaffold a pet with a name and age attributes (hint:
 Also think of the data types for each attribute)?
 
 ```bash
-// your response here
+rails generate scaffold Pet name: string age: integer
```

Comment:
Good couple things though, we can do pet not capitalized. Also don't put a space between name:string and age:integer.